lib/mockito.dart

```diff
@@ -48,6 +53,119 @@ class Mock {
   String toString() => _givenName != null ? _givenName : runtimeType.toString();
 }
 
+// Return a new [Invocation], reconstituted from [invocation], [_typedArgs],
+// and [_typedNamedArgs].
+Invocation _reconstituteInvocation(Invocation invocation) {
+  var newInvocation = new FakeInvocation(invocation);
+  _typedArgs.clear();
+  _typedNamedArgs.clear();
+  return newInvocation;
+}
+
+/// An Invocation class with a functional constructor.
+class FakeInvocation extends Invocation {
+  final Symbol memberName;
```

Comment:
> Can you cite something regarding this?

I remember seeing an announcement or something about that but I can't find it anymore. I could only find https://github.com/dart-lang/sdk/issues/24928#issuecomment-157472898

> Do I need the `@override` annotations on such fields?

You never need `@override` annotations but it's good practice when you are overriding something. In this case these fields are overriding the getters on the extended class, so why not make that explicit?